pkg/resource/label.go

```diff
@@ -1,12 +1,44 @@
 package resource
 
-type NameGetter interface {
+// CRGetter is an interface which contains getter functions
+// to retrieve data from the custom resource.
+type CRGetter interface {
```

Comment:
@tinakurian Where have you implemented `CRGetter` interface ?

Reply:
@tinakurian Function names don't really convey that it's being used to get labels. How about having `GetLabelComponent`, `GetLabelInstance`, `GetLablePartOf` ? Reason I thought of this because when we implement this in any CR type it would be easier to understand this function's significance. 
What do you think ?

Reply:
@akashshinde Thanks for the review! Updated it here: 3ed6948ce43c157b834d94c91544c0a7f8abe936 and here: https://github.com/redhat-developer/devconsole-api/pull/17